cf/src/main/scala/com/timeout/scalacloudformation/Parameter.scala

```diff
@@ -0,0 +1,82 @@
+package com.timeout.scalacloudformation
+import enum.Enum
+
+sealed trait Parameter extends HasLogicalId {
+  def logicalId: String
+  def Type: Parameter.DataType
+  def Description: Option[String]
+  def NoEcho: Option[Boolean]
+}
+
+object Parameter {
+  sealed trait AwsType
+  object AwsType {
+    case object `AWS::EC2::Instance` extends AwsType
+    case object `AWS::EC2::AvailabilityZone::Name` extends AwsType
+    case object `AWS::EC2::Instance::Id` extends AwsType
+    case object `AWS::EC2::Image::Id` extends AwsType
+    case object `AWS::EC2::KeyPair::KeyName` extends AwsType
+    case object `AWS::EC2::SecurityGroup::GroupName` extends AwsType
+    case object `AWS::EC2::SecurityGroup::Id` extends AwsType
+    case object `AWS::EC2::Subnet::Id` extends AwsType
+    case object `AWS::EC2::Volume::Id` extends AwsType
+    case object `AWS::EC2::VPC::Id` extends AwsType
+    case object `AWS::Route53::HostedZone::Id` extends AwsType
+
+    implicit val enum: Enum[AwsType] = Enum.derived[AwsType]
+  }
+
+  sealed trait DataType
+  object DataType {
+    case object String extends DataType
+    case object Number extends DataType
+    case object `List<Number>` extends DataType
+    case object CommaDelimitedList extends DataType
+    case class AwsType(tpe: Parameter.AwsType) extends DataType
+    case class AwsListType(tpe: Parameter.AwsType) extends DataType
```

Comment:
I think this should be called `AwsTypeList` instead. Also, I don't know if it's accurate, as only a subset of the AWS types are available as list items